hydra-node/src/Hydra/Chain/CardanoClient.hs

```diff
@@ -245,69 +273,92 @@ queryEpochNo networkId socket queryPoint = do
 --
 -- Throws at least 'QueryException' if query fails.
 queryProtocolParameters ::
+  -- | Current network discriminant
   NetworkId ->
+  -- | Filepath to the cardano-node's domain socket
   SocketPath ->
   QueryPoint ->
   IO (PParams LedgerEra)
 queryProtocolParameters networkId socket queryPoint = do
-  let query =
-        QueryInEra
-          BabbageEraInCardanoMode
-          ( QueryInShelleyBasedEra
-              ShelleyBasedEraBabbage
-              QueryProtocolParameters
-          )
-  runQuery networkId socket queryPoint query >>= throwOnEraMismatch
+  AnyCardanoEra era <- queryCurrentEra networkId socket QueryTip
+  mkQueryInEra era QueryProtocolParameters
```

Comment:
Could: Query in only one connection to the node by using `queryCurrentEraExpr`

Currently this leads to two connections (line 283 and 285)